### PR TITLE
Cluwne Beast

### PR DESCRIPTION
**Pull request by brainfood1183**
Adds a new creature, the cluwne beast. The cluwne beast has a chance to cluwne any humanoid it hits, turning them into cluwnes. This is an antagonist that should be limited to admin only events and uses the ventcritter event to spawn 1 cluwne beast in a random location on the station. Essentially it is a fun way to have cluwnification be contagious for those times when the admins want to have a little fun.

untitled-300

https://github.com/space-wizards/space-station-14/assets/113240905/20ddd815-c3aa-4295-9b34-6b428219854e

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

Added - cluwnebeast.
Added- honk cloak.

:cl:
- add: Cluwne Beast has been added as an admin event. Honk!

### CODE REVIEW

File: Resources/Audio/Voice/Cluwne/attributions.yml [hunk 1]
```diff
@@ -5,3 +5,9 @@
   license: "CC-BY-4.0"
   copyright: "Evil Laughs 3 Pack by Nanakisan. Modified by brainfood1183 (Github), shortened, converted from WAV to OGG and downmixed to 1 channel."
   source: "https://freesound.org/people/Nanakisan/packs/15544/"
+
+- files:
+    - cluwnebeastlaugh1.ogg
+  license: "CC-BY-4.0"
```
Comment (metalgearsloth) [inline Resources/Audio/Voice/Cluwne/attributions.yml:11]: It's CC BY 3.0

File: Resources/Audio/Misc/attributions.yml [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+- files: ["emergency_meeting.ogg"]
+  license: "CC-BY-NC-SA-3.0"
+  copyright: "Taken from TG station."
+  source: "https://github.com/tgstation/tgstation/commit/97945e7d08d1457ffc27e46526a48c0453cc95e4"
+
+- files: ["beast.ogg"]
+  license: "CC-BY-NC-SA-3.0"
+  copyright: "Partially taken from TG station and made by maximej (freesound.org) and then mixed together, converted to mono and shortened by brainfood1183 (github)."
+  source: "https://github.com/tgstation/tgstation/commit/3e34eb9e3fec7ff55c2db11ba2e7d3b4156443fa"
+  source: "https://freesound.org/people/maximej/sounds/596530/"
```
Comment (metalgearsloth) [inline Resources/Audio/Misc/attributions.yml:10]: Don't mix sources like this

File: Resources/Audio/Misc/attributions.yml [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+
+- files: ["tension_session.ogg, salvage.ogg"]
```
Comment (metalgearsloth) [inline Resources/Audio/Misc/attributions.yml:12]: ?

File: Resources/Audio/Misc/attributions.yml [hunk 4]
```diff
@@ -1,4 +1,15 @@
-- files: ["tension_session.ogg"]
+- files: ["emergency_meeting.ogg"]
+  license: "CC-BY-NC-SA-3.0"
+  copyright: "Taken from TG station."
+  source: "https://github.com/tgstation/tgstation/commit/97945e7d08d1457ffc27e46526a48c0453cc95e4"
```
Comment (metalgearsloth) [inline Resources/Audio/Misc/attributions.yml:4]: Isn't this an unlicensed sound taken from among us?

File: Content.Shared/Cluwne/CluwneComponent.cs [hunk 5]
```diff
@@ -24,6 +25,39 @@ public sealed class CluwneComponent : Component
...
+    /// The autoemote sound to play.
+    /// </summary>
+    [DataField("autoEmoteSound")]
+    public readonly string AutoEmoteSound = "CluwneGiggle";
+
+    /// <summary>
+    /// If this is true then has a % chance to transform target into a cluwne when hit with melee.
+    /// </summary>
+    [DataField("cluwneOnMelee")]
+    public bool CluwneOnMelee = false;
```
Comment (metalgearsloth) [inline Content.Shared/Cluwne/CluwneComponent.cs:47]: This just doubles the above field when you could set it to 0 and get the same effect.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Audio/Misc/attributions.yml (hunk 4 above)
Comment (brainfood1183) [reply Resources/Audio/Misc/attributions.yml:4]: not sure why this was added tbh, i dont need it and will remove.

File: Resources/Audio/Misc/attributions.yml (hunk 3 above)
Comment (brainfood1183) [reply Resources/Audio/Misc/attributions.yml:12]: If i mixed the sound from 2 different sources what do i use for new sound? i plan to remove this sound anyway though.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Some of the hardcoding around cluwne also needs fixing (e.g. component.IsCluwne when this should be dona via yaml not C#).

**Comment by metalgearsloth** [comment]
@brainfood1183 are you still working on this.

**Comment by brainfood1183** [comment]
> @brainfood1183 are you still working on this.

yea, i plan to work on it this weekend.

**Review by lzk228 — CHANGES_REQUESTED** [review]
needs to be made so that cluwne beast can only wear his cloak, because other cloaks do not have a texture for him

File: Content.Server/Cluwne/CluwneSystem.cs [hunk 6]
```diff
@@ -44,14 +47,20 @@ public override void Initialize()
     /// </summary>
     private void OnMobState(EntityUid uid, CluwneComponent component, MobStateChangedEvent args)
     {
-        if (args.NewMobState == MobState.Dead)
-		{
+
+        if (args.NewMobState == MobState.Dead && component.IsCluwne == false)
```
Comment (Slava0135) [inline Content.Server/Cluwne/CluwneSystem.cs:51]:
```suggestion
        if (args.NewMobState == MobState.Dead)
        {
                return;
        }
        RemComp<AutoEmoteComponent>(uid);
```

File: Content.Server/Cluwne/CluwneSystem.cs [hunk 7]
```diff
@@ -44,14 +47,20 @@ public override void Initialize()
...
+        {
+            RemComp<AutoEmoteComponent>(uid);
+        }
+
+        if (args.NewMobState == MobState.Dead && component.IsCluwne)
```
Comment (Slava0135) [inline Content.Server/Cluwne/CluwneSystem.cs:56]:
```suggestion
        if (component.IsCluwne)
```

File: Content.Server/Cluwne/CluwneSystem.cs [hunk 8]
```diff
@@ -69,15 +78,26 @@ private void OnComponentStartup(EntityUid uid, CluwneComponent component, Compon
...
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cluwne-transform", ("target", uid)), uid, PopupType.LargeCaution);
+            _audio.PlayPvs(component.SpawnSound, uid);
+            _metaData.SetEntityName(uid, Loc.GetString("cluwne-name-prefix", ("target", name)), meta);
+            SetOutfitCommand.SetOutfit(uid, "CluwneGear", EntityManager);
+            _npcFaction.RemoveFaction(uid, "NanoTrasen", false);
+            _npcFaction.AddFaction(uid, "HonkNeutral");
 
-        _metaData.SetEntityName(uid, Loc.GetString("cluwne-name-prefix", ("target", name)), meta);
+        }
```
Comment (Slava0135) [inline Content.Server/Cluwne/CluwneSystem.cs:94]:
```suggestion
```

File: Content.Server/Weapons/Melee/InfectOnMelee/InfectOnMeleeSystem.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Weapons.Melee.Events;
+using Content.Shared.Zombies;
+using Content.Shared.Weapons.Melee;
+
+namespace Content.Server.Weapons.Melee.InfectOnMelee;
+
+public sealed class InfectOnMeleeSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
```
Comment (Slava0135) [inline Content.Server/Weapons/Melee/InfectOnMelee/InfectOnMeleeSystem.cs:14]:
```suggestion
    [Dependency] private readonly IRobustRandom _random = default!;
```

File: Content.Shared/Cluwne/CluwneComponent.cs [hunk 10]
```diff
@@ -24,6 +25,27 @@ public sealed partial class CluwneComponent : Component
     [DataField("emoteId", customTypeSerializer: typeof(PrototypeIdSerializer<EmoteSoundsPrototype>))]
     public string? EmoteSoundsId = "Cluwne";
 
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float Cluwinification = 0.15f;
```
Comment (Slava0135) [inline Content.Shared/Cluwne/CluwneComponent.cs:29]: doesn't seem to do anything
```suggestion
```

File: Content.Shared/Cluwne/CluwneComponent.cs [hunk 11]
```diff
@@ -24,6 +25,27 @@ public sealed partial class CluwneComponent : Component
...
+    /// The autoemote sound to play.
+    /// </summary>
+    [DataField("autoEmoteSound")]
+    public string AutoEmoteSound = "CluwneGiggle";
+
+    /// <summary>
+    /// Portal proto id.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("portal", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string Portal = "PortalGreeny";
```
Comment (Slava0135) [inline Content.Shared/Cluwne/CluwneComponent.cs:47]:
```suggestion
    public EntProtoId Portal = "PortalGreeny";
```

File: Content.Shared/Cluwne/CluwneComponent.cs [hunk 12]
```diff
@@ -24,6 +25,27 @@ public sealed partial class CluwneComponent : Component
...
+    /// <summary>
+    /// The autoemote sound to play.
+    /// </summary>
+    [DataField("autoEmoteSound")]
+    public string AutoEmoteSound = "CluwneGiggle";
+
+    /// <summary>
+    /// Portal proto id.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("portal", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
```
Comment (Slava0135) [inline Content.Shared/Cluwne/CluwneComponent.cs:46]:
```suggestion
    [ViewVariables(VVAccess.ReadWrite), DataField("portal")]
```

File: Content.Shared/Weapons/Melee/Components/InfectOnMeleeComponent.cs [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+namespace Content.Shared.Weapons.Melee;
+
+[RegisterComponent]
+[NetworkedComponent]
+public sealed partial class InfectOnMeleeComponent : Component
+{
+    /// <summary>
+    /// infection chance determines the % chance that target will be infected.
+    /// </summary>
+
```
Comment (Slava0135) [inline Content.Shared/Weapons/Melee/Components/InfectOnMeleeComponent.cs:13]:
```suggestion
```
remove spaces between the thing and comment (i think)

**Review by Slava0135 — CHANGES_REQUESTED** [review]

File: Content.Server/Cluwne/CluwneSystem.cs [hunk 14]
```diff
@@ -45,13 +48,12 @@ public override void Initialize()
     private void OnMobState(EntityUid uid, CluwneComponent component, MobStateChangedEvent args)
     {
         if (args.NewMobState == MobState.Dead)
-		{
-            RemComp<CluwneComponent>(uid);
+        {
+            var damageSpec = new DamageSpecifier(_proto.Index<DamageGroupPrototype>("Genetic"), 300);
+            _damageable.TryChangeDamage(uid, damageSpec);
             RemComp<ClumsyComponent>(uid);
-            RemComp<AutoEmoteComponent>(uid);
```
Comment (Slava0135) [inline Content.Server/Cluwne/CluwneSystem.cs:51]: why you don't remove AutoEmote component, won't this make them giggle when dead?

Comment (brainfood1183) [reply Content.Server/Cluwne/CluwneSystem.cs:51]: doesnt make em giggle when dead cause they require cluwnecomponent to giggle.

File: Content.Shared/Weapons/Melee/Components/InfectOnMeleeComponent.cs [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    /// infection chance determines the % chance that target will be infected.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float InfectionChance = 0.15f;
+
+    /// <summary>
+    /// If this is true then target will be cluwned.
+    /// </summary>
+    [DataField("cluwnification")]
+    public bool Cluwinification = false;
```
Comment (metalgearsloth) [inline Content.Shared/Weapons/Melee/Components/InfectOnMeleeComponent.cs:20]: 1. Infection needs to be a datafield and networked.
2. This should not be tied to cluwne and zombies needs to use it.

File: Content.Server/Weapons/Melee/InfectOnMelee/InfectOnMeleeSystem.cs [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<InfectOnMeleeComponent, MeleeHitEvent>(OnMeleeHit);
+    }
+
+    private void OnMeleeHit(EntityUid uid, InfectOnMeleeComponent component, MeleeHitEvent args)
+    {
+        if (component.Cluwinification == true)
+        {
```
Comment (metalgearsloth) [inline Content.Server/Weapons/Melee/InfectOnMelee/InfectOnMeleeSystem.cs:28]: Please invert the condition to remove one level of indentation and you also don't need to check if it's true as per C# style-guide.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Cluwne is still very singleton-like in terms of code and needs to be more generic.

A lot of the code just doubles zombies when 95% of it should be shared between the same systems with minor differences.

**Comment by chromiumboy** [comment]
Hey there @brainfood1183, are you still working on this PR?

**Comment by brainfood1183** [comment]
> Hey there @brainfood1183, are you still working on this PR?

not currently, can probably be closed for now and maybe reopened later.

**Comment by chromiumboy** [comment]
No worries, you can always re-open this if you decide to come back to it later

**Pull request closed without merging**